### ISSUE TITLE
Authentication Succesful But Nothing Happens

### ISSUE THREAD

**Issue by Time2Veto** [KRTirtho/spotube#56]
So I can put in the ClientID and the SecretID and then it'll bring me to the proper authorize access page BUT then it gets stuck on successful with nothing changing in the app. Sits on the submit page where I keep opening the allow access page. I am having this issue on Android and Windows.

**Expected behavior**
Should Log me in.

**Desktop:**
 - OS: Windows 10
 - Version: Latest

**Smartphone (please complete the following information):**
 - Device: Pixel 6 Pro
 - OS: Android
 - Version: Lastest

**Comment by KRTirtho** [comment]
Close the app. In android, Clear Storage of the app from the app info & try again

**Comment by Time2Veto** [comment]
I tried that just now. Nothing changed just sits on the successful screen. No change in app.

**Comment by Time2Veto** [comment]
It doesn't change the outcome. Just gets stuck at this page and doesn't sign me in. 
![Uploading Screenshot_20220406-235355.png…]()

**Comment by KRTirtho** [comment]
The screenshot redirects to the issue. Provide correctly

**Comment by Time2Veto** [comment]
Sorry trying to. Mobile is hating me right now. [https://imgur.com/a/QbPlanc](url)

**Comment by KRTirtho** [comment]
Again wrong URL

**Comment by Time2Veto** [comment]
I'm not sure what exactly I'm doing wrong about it? It's the right if you copy and paste it. Just clicking the hyperlink redirects to this page for some reason.

**Comment by KRTirtho** [comment]
Can provide the app screenshot?

**Comment by Time2Veto** [comment]
Sure. So yeah. 
1. Enter in info hit submit
2. Get the success page I showed you
3. Nothing happens until I go back to app 
4. Nothing changes on app. Like I never hit submit [](https://imgur.com/TkfjGRL)

**Comment by KRTirtho** [comment]
Try hitting back & see if a red logout button appears or if its still showing a green connect with Spotify button

**Comment by Time2Veto** [comment]
That's what I try. Still green and no login at all.

**Comment by KRTirtho** [comment]
Did you try uninstalling & reinstalling the app

**Comment by Time2Veto** [comment]
Just tried and no changes. Still doesn't work.

**Comment by Time2Veto** [comment]
So just an update. Been trying daily and still nothing. Maybe its my phone? Or account?

**Comment by KRTirtho** [comment]
It could be a special case for your phone or account. Why not try with a new account?

**Comment by Time2Veto** [comment]
I'd imagine it must be my account. Just shame if it is.

**Comment by Time2Veto** [comment]
It's my account. So I have no idea why but my original account can't sign in. Wonderful. Any ideas or just lost cause? Only cuz it's recommend is better catered

**Comment by KRTirtho** [comment]
Are you using Spotify premium?

**Comment by Time2Veto** [comment]
Would that cause an issue? I actually am cuz I paid for it thinking your app would block ads lol I know it does now but at the time.

**Comment by KRTirtho** [comment]
Actually, I didn't test the app with premium accounts so it could be a reason causing error in the app

**Comment by Time2Veto** [comment]
Well I can try it out when it runs out next month

**Comment by KRTirtho** [comment]
Yep, that would be helpful

**Comment by karniv00l** [comment]
I've tried both macos and android 11 with premium account, works fine for me

**Comment by KRTirtho** [comment]
So I guess its an edge case for @Time2Veto's account

**Comment by Time2Veto** [comment]
The only thing I can think of is this being an issue with it being like a hulu/spotify student bundle that ran out? Like this is from years ago so maybe somethings different about it. Last idea. Not a big deal though

**Comment by senolfndk** [comment]
not logged in ![Ekran görüntüsü 2022-04-25 175328](https://user-images.githubusercontent.com/52818836/165115429-d40bbb18-35f3-4234-91ff-111462f8116d.png)

**Comment by Gottarocket** [comment]
Authentications successful on my end too but no redirection to app. Not premium . Changed browser still same issue. I believe its because its old account or is it location issue?

**Comment by PranavBhattarai** [comment]
Guys the problem is, after when we give Spotify permission to Spotube, the browser tries to go to a URL named "**localhost:4304/auth/spotify/callback /....**" which on PC (either in Windows or Linux) **does** redirect to Spotube App but when it comes to mobile (Android), **it doesn't**. 

I wonder, has anyone done this on their mobile successfully! How does the **mobile browser** redirect "the successful token" to **application** installed on the phone? This logic needs some fixing to do.

**Comment by 92flash** [comment]
I have the same problem too.

**Comment by KRTirtho** [comment]
@Gottarocket _"Authentication Successful"_ means you're authenticated with your Spotify Account in Spotube. The browser won't redirect to the app even if the authentication is successful. You've to manually go to the app & check if you're logged in. If everything went fine you'll see a Red _"Log Out"_ button on the _Settings Page_ of Spotube

**Comment by senolfndk** [comment]
😉
![Ekran görüntüsü 2022-05-22 082744](https://user-images.githubusercontent.com/52818836/169680346-adc6cc2b-8772-41c8-b50d-996dd445b695.png)
![kokok](https://user-images.githubusercontent.com/52818836/169680347-aeaa4503-955d-4595-a34d-8ee099dd92c3.png)
![olko](https://user-images.githubusercontent.com/52818836/169680348-3fc19b87-1c16-42a1-b226-b1952e7df9a1.png)

**Comment by Gottarocket** [comment]
Android app . It was where I have issue . Sorry if it has separate repo to file issue.

**Comment by KRTirtho** [comment]
@Gottarocket The authentication procedure is same for all platforms (Android, Linux, Windows, Mac)

> Android app . It is where I have issue . Sorry if it has separate repo to file issue.
Spotube is using single codebase for all platforms so issues of every platform can be filed in the same repo

**Comment by Gottarocket** [comment]
Sure . On mobile open in app option should be available as authentication successful doesn't update on app end so it still asks to login

**Comment by PranavBhattarai** [comment]
@karniv00l 
I think we can really close this issue. Because after the update of ver 2.2.0, I was able to successfully login in SpotTube. It's a bit tricky though:
1. Install the latest Spotube (if you have not installed the new version yet)
2. Try to login into Spotify using its credentials
3. It will redirect you to your default browser (in my case, it's Firefox ver.102.2)
4. It can/will throw an error, like "some went wrong"... or something... when you give Spotube the permission from Spotify website.
5. Select desktop mode, to view that page, which will again ask permission to allow Spotube.
6. Spotify website will give you a response saying, "Successfully gained permission, return to Spotube".
6.1 But when you go back to Spotube, it doesn't show that you are already logged in.
7. Repeat Steps 2, 3, 4, 5, and 6 one more time, and _voila`_ you have logged in for real.

The important thing is: You need to log into Spotify, at least **twice**. I know its sounds weird but totally works.

Tested Device: Oneplus 5
ROM: Pixel Experience 12 (android 12 L)

The only annoying thing when it comes to **Android** is this: 
```
you get an update notification 
  you download the latest update from GitHub 
   you try to install that package
    it will throw an error saying, "packaging error"
     you uninstall previous Spotube installed on your phone
     you install the new version
      repeat the painful cycle of Step 1 to Step 7 again. 
```
Really kills the vibe doing this on REPEAT.

**Comment by KRTirtho** [comment]
@PranavBhattarai the login was successful in the very 1st time. You just had to check whether you can access search & library tab or not. If you can then you're logged in

**Comment by PranavBhattarai** [comment]
@karniv00l  Maybe I or maybe I didn't check the library. But the important thing was during the 1st attempt, Spotube didn't show the **Logout** option in the place of the Spotify option and it was only shown after the **second attempt*.

I will carefully notice the library tab after the 1st attempt again in the next Spotube update (since I have to redo this thing again).

**Comment by KRTirtho** [comment]
It should be fixed in 2.4.0. Can you guys check again?

**Comment by PranavBhattarai** [comment]
yeah, it works. Thank you. Lovely update. Looks sleeker.

**Comment by PranavBhattarai** [comment]
@KRTirtho 
![spot](https://user-images.githubusercontent.com/32763112/189479470-2e9f7ba9-5d2a-4634-9ddc-c5ebacfc4544.jpg)
```
Device Tested: Oneplus 5
ROM: Android 12L
```

**Comment by KRTirtho** [comment]
@PranavBhattarai 
Can you,
- connect your device via ADB
- run `flutter logs`
- start using the app
- see what logs it outputs
- finally provide the logs here?

**Comment by PranavBhattarai** [comment]
@KRTirtho I wish I could. I USB port of this phone is a _bit_ damaged _(I think)_. 
I haven't been able to transfer files or use the ADB command in ages. 
The phone is in "Charge only mode". Only charge. No data transfer.

**Comment by KRTirtho** [comment]
v2.5.0 fixed the issue. Download & install the latest version :smile:

**Comment by KRTirtho** [comment]
duplicate of #152

**Comment by senolfndk** [comment]
> v2.5.0, sorunu düzeltti. En son sürümü indirin ve yükleyin😄

ok

**Comment by KRTirtho** [comment]
Closing this issue as it was fixed in v2.5.0
